### DISCUSSION TITLE
emcee_solver: errors with requivsumfrac

### DISCUSSION THREAD

**Discussion by hvn7** [phoebe-project/phoebe2#816] [Questions]
Hello!

I am trying to run the emcee solver to get the uncertainties, but keep running into this error saying: 

BUNDLE       ERROR   Constraint 'requiv@primary@star@constraint' raised the following error while setting the value of 'requiv@primary@star@component'.  Original error: value of requiv=-0.28210775244503 solRad not within limits of [<Quantity 1.e-06 solRad>, None].
For every iteration I get the same error with different "values of requiv".

The codes is as follow:

```
b.add_compute('ellc', compute='fastcompute',overwrite=True)
b.set_value_all('ld_mode', 'manual')
b.set_value_all('ld_mode_bol','manual')
b.set_value_all('atm','extern_planckint')
b.set_value_all('pblum_mode','dataset-scaled')
pblums_scaled = b.compute_pblums(compute='fastcompute', model='after_estimators')
print(pblums_scaled)

b.set_value_all('pblum_mode', 'component-coupled')
b.set_value('pblum', dataset='lc01', component='primary', value=pblums_scaled['pblum@primary@lc01'])
print(b.compute_pblums(compute='fastcompute', dataset='lc01', pbflux=True))
b.flip_constraint('sma@binary', solve_for='asini')
b.add_distribution({'teffratio': phoebe.gaussian_around(0.001),
                    'requivsumfrac': phoebe.gaussian_around(0.1),
                    'incl@binary': phoebe.gaussian_around(3),
                    'sma@binary': phoebe.gaussian_around(2),
                    'q': phoebe.gaussian_around(0.01),
                    'ecc': phoebe.gaussian_around(0.01),
                    'per0': phoebe.gaussian_around(5),
                    'pblum': phoebe.uniform_around(0.000001)},
                    distribution='ball_around_optimized_solution')

_ = b.plot_distribution_collection('ball_around_optimized_solution', show=True)

b.add_solver('sampler.emcee',
             init_from='ball_around_optimized_solution',
             compute='fastcompute',
             solver='emcee_solver')
b.run_solver('emcee_solver', niters=300, nwalkers=16, solution='emcee_sol')
```

I tried to change the values for the b.add_distribution step but this error keeps showing up. I have attached some pictures for my codes as well. Can anyone help me with this issue, please?
Thank you very much.
<img width="1140" alt="Screenshot 2023-12-21 at 12 39 19" src="https://github.com/phoebe-project/phoebe2/assets/154519076/e156c285-ea17-4425-9fee-681f5c8dab62">
<img width="1137" alt="Screenshot 2023-12-21 at 12 39 39" src="https://github.com/phoebe-project/phoebe2/assets/154519076/1f4b45ad-b872-4a60-b882-35fd3fe65b8e">
<img width="1140" alt="Screenshot 2023-12-21 at 12 39 53" src="https://github.com/phoebe-project/phoebe2/assets/154519076/7b5624bc-e84d-4ffe-99a5-41dade32028b">
<img width="1139" alt="Screenshot 2023-12-21 at 12 40 09" src="https://github.com/phoebe-project/phoebe2/assets/154519076/0f9f96f9-5f92-4cc9-bda5-33e429106ec2">
<img width="1137" alt="Screenshot 2023-12-21 at 12 47 19" src="https://github.com/phoebe-project/phoebe2/assets/154519076/ddf0e433-327c-45e1-bc59-a8dffbddecca">
<img width="1136" alt="Screenshot 2023-12-21 at 12 47 29" src="https://github.com/phoebe-project/phoebe2/assets/154519076/3dba090f-9142-4a27-baa2-52a25f139de9">
<img width="1134" alt="Screenshot 2023-12-21 at 12 47 39" src="https://github.com/phoebe-project/phoebe2/assets/154519076/76757345-b3e4-4ee0-baa8-0d6c81156c41">
<img width="1143" alt="Screenshot 2023-12-21 at 12 47 48" src="https://github.com/phoebe-project/phoebe2/assets/154519076/6cd766b6-8d43-4f60-89d8-d5641f80ccd3">
<img width="1145" alt="Screenshot 2023-12-21 at 12 47 57" src="https://github.com/phoebe-project/phoebe2/assets/154519076/f67aae3c-2d1f-4391-916b-586dcefd4146">

(For further context, I was not able to perform Optimizer so I am working with the data I got from the Estimators)

**Comment by kecnry** [comment]
These error messages aren't a major concern on their own - whenever they happen, that sample will be rejected.  However, I think this is likely happening more than necessary because your initializing distributions contain a lot of non-physical parameter combinations.  There is a possibility that you even have walkers that never find their way into physical parameter space, which might explain the "islands" you're seeing in the corner plot.  You have a few options to try to overcome this:
* have [phoebe continue resampling the initial position until they all start in physical parameter space](http://phoebe-project.org/docs/latest/tutorials/emcee_init_from_requires).  This is probably the simplest to do, but does add quite a bit expense, so you might want to consider in combination with one of the other options as well.
* tighten your distributions to minimize how many samples fall outside physical bounds
* include uniform distributions on some/all of the distributions using and logic.  To do this, create another distribution collection with the uniform "bounds", set `init_from` to be a list of both your gaussian and uniform distributions, and set `init_from_combine` to `"and"`.

> For further context, I was not able to perform Optimizer so I am working with the data I got from the Estimators

What is giving you problems with the optimizer step?  Sampling will be quite computationally expensive if you're not already pretty tightly confined to the eventual "solution".
